source/common/http/filter_manager.cc

```diff
@@ -1064,6 +1064,19 @@ void FilterManager::encodeHeaders(ActiveStreamEncoderFilter* filter, ResponseHea
     }
   }
 
+  const auto status = HeaderUtility::checkRequiredResponseHeaders(headers);
```

Comment:
not part of the diff but at line 1014 there's an ASSERT on status and you may want to add that the headers will be valid at this point since they've just finishing parsing and haven't had any modification.